packages/crypto/src/symmetric.ts

```
@@ -37,6 +37,78 @@ const decryptBytes = (
+
+  const createEncryptStream = async function*(stream: AsyncIterable<Uint8Array>, state: StateAddress): AsyncGenerator<Uint8Array> {
+    for await (const chunk of stream) {
+      const encryptedChunk = sodium.crypto_secretstream_xchacha20poly1305_push(
+        state,
+        chunk,
+        null,
+        sodium.crypto_secretstream_xchacha20poly1305_TAG_MESSAGE
+      );
+      yield encryptedChunk
+      
+    }
+
+    // Finalize the stream
+    const encryptedFinalChunk = sodium.crypto_secretstream_xchacha20poly1305_push(
+      state,
+      new Uint8Array(),
+      null,
+      sodium.crypto_secretstream_xchacha20poly1305_TAG_FINAL
+    );
+    yield encryptedFinalChunk
+  }
+
+  return {
+    encryptStream: createEncryptStream(stream, state),
+    header
+  }
+}
+
+const decryptBytesStream = (
+  /** The stream of encrypted bytes to decrypt */
+  encryptedStream: AsyncIterable<Uint8Array>,
+  /** Header bytes written to the encrypted stream */
+  header: Uint8Array,
+  /** The password used to encrypt */
+  password: Password
+): AsyncGenerator<any> => {
+  const key = stretch(password)
+
+  const state = sodium.crypto_secretstream_xchacha20poly1305_init_pull(header, key);
+  console.log(state, typeof state)
+
+  const createDecryptStream = async function*(encryptedStream: AsyncIterable<Uint8Array>, state: StateAddress): AsyncGenerator<any> {
+    for await (const chunk of encryptedStream) {
+      const decryptedChunk = sodium.crypto_secretstream_xchacha20poly1305_pull(
+        state,
+        chunk
+      );
+      // all valid encrypted chunks on the stream should end with this tag
+      if (decryptedChunk.tag == sodium.crypto_secretstream_xchacha20poly1305_TAG_MESSAGE) {
+        yield decryptedChunk.message
+      } else if (decryptedChunk.tag != sodium.crypto_secretstream_xchacha20poly1305_TAG_FINAL) {
+        console.error(`Unknown tag seen while decrypting stream`, decryptedChunk.tag)
```

Should we just throw an error if we get a PUSH or REKEY tag? REKEY would cause us to be unable to decrypt the rest of the stream if we didn't handle it, and we don't handle it in our protocol. Probably best to actually throw an error so that the enclosing function can handle the stream being incomplete